script.js

```diff
@@ -1,4 +1,338 @@
+////////////////////////////
+// Helper functions
+var makeDeck = function () {
+  var cardDeck = [];
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      cardDeck.push(card);
+
+      rankCounter += 1;
+    }
+
+    suitIndex += 1;
+  }
+
+  return cardDeck;
+};
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+var cardDeck = shuffleCards(makeDeck());
+
+///////////////////////////////////////////////
+// State/Global variables
+// constant winning value of blackjack
+var twentyOne = 21;
+// Dealer will always submit "hit" if its sum is below 16
+var dealerHitThreshold = 16;
+// If player has submitted "stand", player can no longer submit "hit", until game is over
+var playerHasChosenToStand = false;
+// Keep track of when the game ends to prevent further moves (i.e., "hit" and "stand")
+var gameOver = false;
+
+var playerHand = [];
+var computerHand = [];
+
+var outputDiv = document.getElementById("output-div");
+var divContainer = document.getElementById("container");
+
+var myImage =
+  '<img src = "https://media.tenor.com/No8u0Yip0lwAAAAC/so-pissed.gif"/>';
+
+var myImage2 =
+  '<img src = "https://media.tenor.com/kfJ-3t3IFdgAAAAM/1inch-1inch-exchange.gif"/>';
+
+var myImage3 = `<img src = "https://media.tenor.com/cxsA-a-8uz0AAAAC/tom-and-jerry-jerry-the-mouse.gif" />`;
+
+///////////////////////////////
+// More helper functions
+
+// Deal a card to the given hand
+var dealCardToHand = function (hand) {
```

Comment:
this is a nice function. But you can take it one step further by accepting a second argument to determine how many cards to deal